salt/modules/postgres.py

```diff
@@ -886,7 +886,7 @@ def get_bool(rowdict, key):
         for date_key in ('expiry time',):
             try:
                 retrow[date_key] = datetime.datetime.strptime(
-                    row['date_key'], '%Y-%m-%d %H:%M:%S')
+                    row[date_key], '%Y-%m-%d %H:%M:%S')
```

Comment:
I agree with your analysis here. IMHO, the `KeyError` handling should be safe to remove.